sklearn/utils/validation.py

```diff
@@ -420,6 +424,13 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
         The converted and validated array.
 
     """
+    # warn_on_dtype deprecation
+    if warn_on_dtype is not None:
+        warnings.warn(
+            "'warn_on_dtype' is deprecated in "
+            "version 0.21 has no effect and will be removed in 0.23.",
```

Comment:
This message is wrong.

warn_on_dtype should have an effect since it is deprecated, not removed

Reply:
done